include/posix/unistd.h

```diff
@@ -11,11 +11,46 @@ extern "C" {
 #endif
 
 #include "sys/types.h"
+#include "sys/stat.h"
+
+#ifdef CONFIG_POSIX_FS
+#include <fs.h>
+
+#undef PATH_MAX
```

Comment:
`PATH_MAX` should be defined in `limits.h`, not `unistd.h`.  Redefining it here will mean that, if some code uses `limits.h` and not `unistd.h`, and it interacts with some code that uses `unistd.h`, they'll most likely be using different values.

`PATH_MAX` can be redefined only if it's being redefined to, say, `CONFIG_PATH_MAX`, so that we can make sure that it's the same value everywhere in the system, while making it configurable to boot (which is a good thing: a `struct dirent` might be allocated on the stack, and 256 bytes is too much for the tiny devices; in fact, that's as much kernel stack ARM devices have to handle syscalls).

Reply:
In limits.h (usr/incude/limits.h) is set as 4096 bytes so I have create another header file like "sys/type.h" for limits.h. For now, we keep in unistd.h. I will do the needed changes along with the next set of requirements.

Reply:
This is a security vulnerability waiting to happen.